src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/TripleDES.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public abstract class TripleDES : SymmetricAlgorithm
+    {
+        protected TripleDES()
+        {
+            KeySize = 3*64;
+            BlockSize = 64;
+        }
+
+        public static TripleDES Create()
+        {
+            throw new NotImplementedException(SR.WorkInProgress);
+        }
+
+        public override KeySizes[] LegalKeySizes
+        {
+            get
+            {
+                return s_legalKeySizes.CloneKeySizesArray();
+            }
+        }
+
+        public override KeySizes[] LegalBlockSizes
+        {
+            get
+            {
+                return s_legalBlockSizes.CloneKeySizesArray();
+            }
+        }
+
+        public override byte[] Key
+        {
+            get
+            {
+                byte[] key = base.Key;
+                while (IsWeakKey(key))
+                {
+                    GenerateKey();
+                    key = base.Key;
+                }
+                return key;
+            }
+
+            set
+            {
+                if (value == null)
+                    throw new ArgumentNullException("value");
+
+                if (!(value.Length*8).IsLegalSize(s_legalKeySizes))
```

Comment:
I personally don't like us having domain-specific extension methods on integers like this, but I realize you're not introducing this as part of your change, so ok.